### PR TITLE
Feat: Pixiu-SpringCloud use Zookeeper

### PR DESCRIPTION
**Pull request by PhilYue**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/adapter/springcloud/cloud.go [hunk 1]
```diff
@@ -18,6 +18,8 @@
 package springcloud
 
 import (
+	"github.com/apache/dubbo-go-pixiu/pkg/adapter/springcloud/servicediscovery/zookeeper"
```
Comment (ztelur) [inline pkg/adapter/springcloud/cloud.go]: use `imports-formatter -path . -module github.com/apache/dubbo-go-pixiu -bl false` to split import block

File: pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+		if diff != nil {
+			logger.Debugf("Del the service %s", diff)
+			for _, sn := range diff {
+				for _, instance := range discovery.getServiceMap()[sn] {
+					discovery.delServiceInstance(instance)
+				}
+			}
+		}
+	}
+	del()
```
Comment (ztelur) [inline pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go:146]: maybe use below format is better 
`
	func() {
		....
	}()
`

File: pkg/remote/zookeeper/client.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		newClient, cltErr := NewZkClient(zkName, config)
+		if cltErr != nil {
+			return perrors.WithMessagef(cltErr, "create zookeeper client (address:%+v)", config.Address)
+		}
+		container.SetZkClient(newClient)
+	}
+	return nil
+}
+
+func NewZkClient(zkName string, config *model.RemoteConfig) (*gxzookeeper.ZookeeperClient, error) {
```
Comment (ztelur) [inline pkg/remote/zookeeper/client.go:53]: is there a zk client in pixiu or gost for reuse ?

File: pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+			continue
+		}
+		failTimes = 0
+		if continueLoop := z.waitEventAndHandlePeriod(children, e); !continueLoop {
+			return
+		}
+	}
+}
+
+func (z *zkAppListener) waitEventAndHandlePeriod(children []string, e <-chan zk.Event) bool {
```
Comment (AlexStocks) [inline pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go]: waitEvent 就可以了，后面这个 HandlePeriod 名字太难理解

File: pkg/adapter/springcloud/cloud.go (hunk 1 above)
Comment (PhilYue) [reply pkg/adapter/springcloud/cloud.go]: cloud.go 我先手动处理，之后再提一个 formatter 的PR，否则改动太多了

File: pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go (hunk 4 above)
Comment (PhilYue) [reply pkg/adapter/springcloud/servicediscovery/zookeeper/application_listener.go]: Done

File: pkg/remote/zookeeper/client.go (hunk 3 above)
Comment (PhilYue) [reply pkg/remote/zookeeper/client.go:53]: 是呢，打算统一使用 gost 的 zk-client

**Review by ztelur — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Review by mark4z — APPROVED** [review]

**Pull request merged**